Add option to use frame instead of timestamp for stinger transition

derrod opened the pull request:
Other applications I have used in the past (e.g. vMix) use the number of the frame to transition on instead of the timestamp within the file, this change enables that behaviour in the new native stinger transition.

I'm not sure if the implementation is the best way to do this, I think it could be improved, especially the way how the number of frames is estimated.

I tested it with two different stinger animations I had and verified that the transition happens at the correct point.

RytoEX commented on plugins/obs-ffmpeg/obs-ffmpeg-source.c:
```diff
@@ -365,6 +365,43 @@ static void get_duration(void *data, calldata_t *cd)
+
+	AVStream *stream = s->media.fmt->streams[video_stream_index];
+
+	if (stream->nb_frames > 0)
```
Style:  When one branch of an `if` statement needs curly braces, use curly braces for all branches.

RytoEX commented on plugins/obs-transitions/transition-stinger.c:
```diff
@@ -24,6 +30,7 @@ static void stinger_update(void *data, obs_data_t *settings)
 	struct stinger_info *s = data;
 	const char *path = obs_data_get_string(settings, "path");
 
+
```
Not sure why this blank line has been added?

derrod commented:
Fixed style issues, there was an empty line where I deleted some temporary code.

Not sure why the appveyor build failed but it seems to be working this time.
